From 42550cb52fa42a728ed254ae4e68c365fe755759 Mon Sep 17 00:00:00 2001
From: Antoine Bouyer <antoine.bouyer@nxp.com>
Date: Fri, 4 May 2018 11:48:09 +0200
Subject: [PATCH 3761/5242] MLK-18261 irqchip: imx-irqsteer: fix idx
 calculation for mask callback

commit  fa63d5a6ab056a93e8f202861fea7f6cf5243e3e from
https://source.codeaurora.org/external/imx/linux-imx.git

Fixes: a2e6a7833495 (MLK-16136-9 irqchip: imx-irqsteer: adjust irq config
via 'endian')

This patch fixes mask register offset calculation, when endian is not
default value 0 (i.e imx8mq).

Signed-off-by: Antoine Bouyer <antoine.bouyer@nxp.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/irqchip/irq-imx-irqsteer.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/irqchip/irq-imx-irqsteer.c b/drivers/irqchip/irq-imx-irqsteer.c
index eec5308..e26d242 100644
--- a/drivers/irqchip/irq-imx-irqsteer.c
+++ b/drivers/irqchip/irq-imx-irqsteer.c
@@ -47,8 +47,8 @@ static void imx_irqsteer_irq_unmask(struct irq_data *d)
 	u32 val, idx;
 
 	spin_lock(&irqsteer_data->lock);
-	idx = irqsteer_data->endian ? (irqsteer_data->channum - d->hwirq / 32 - 1) :
-				      d->hwirq / 32;
+	idx = irqsteer_data->endian ?
+		(irqsteer_data->channum - d->hwirq / 32 - 1) : d->hwirq / 32;
 	reg = irqsteer_data->regs + CHANMASK(idx);
 	val = readl_relaxed(reg);
 	val |= 1 << (d->hwirq % 32);
@@ -63,7 +63,8 @@ static void imx_irqsteer_irq_mask(struct irq_data *d)
 	u32 val, idx;
 
 	spin_lock(&irqsteer_data->lock);
-	idx = d->hwirq / 32;
+	idx = irqsteer_data->endian ?
+		(irqsteer_data->channum - d->hwirq / 32 - 1) : d->hwirq / 32;
 	reg = irqsteer_data->regs + CHANMASK(idx);
 	val = readl_relaxed(reg);
 	val &= ~(1 << (d->hwirq % 32));
-- 
1.7.9.5

